### PR TITLE
Feature/homepage

### LINKED ISSUE #11: View past trip plans
As a user, I would like the ability to view previous trip plans, so that I can see details about that trip.

**Acceptance Criteria:**
- User should be able to view the details of their previously created trip plan
- User should be able to see where they have went to in that trip (destinations and events)
- User should be able to see the dates of where they have been
- User should be able to see the notes tied to the trip (including ones tied to destinations and events)

**Notes:**

### PR DESCRIPTION
**Pull request by niuniuanran**
Once logged in, the user now can see their current, planning and past trips, with new message/note alerts
closes #11

### CODE REVIEW

File: src/spa/components/homepage/TripCard.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import styles from '../../css/trip-card.module.css';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStickyNote, faCommentDots } from '@fortawesome/free-solid-svg-icons';
+import Tooltip from '../Tooltip';
+
+export default class TripCard extends React.Component {
+  render() {
+    const newChatNum =
+      this.props.newChatNum < 1
```
Comment (rafiazman) [inline src/spa/components/homepage/TripCard.js:13]: abstract this small functionality into its own getter class method (e.g `getChatCount()` that returns `number`). the method would still read from `this.props`

File: src/spa/components/homepage/TripCard.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        ? ''
+        : this.props.newChatNum > 99
+        ? '99+'
+        : this.props.newChatNum;
+    const newNoteNum =
+      this.props.newNoteNum < 1
```
Comment (rafiazman) [inline src/spa/components/homepage/TripCard.js:19]: abstract this small functionality into its own getter class method (e.g `getNoteCount()` that returns `number`)

File: src/spa/components/homepage/UserHomePage.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import {
+  faShoePrints,
+  faPen,
+  faClock,
+} from '@fortawesome/free-solid-svg-icons';
+
+export default class UserHomePage extends React.Component {
+  render() {
+    const today = new Date();
+    const currentTrip = myTripInfos.find(tripInfo => {
```
Comment (rafiazman) [inline src/spa/components/homepage/UserHomePage.js:22]: this is backend functionality, assume array is always "correct"

File: src/spa/components/homepage/UserHomePage.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      return (
+        today >= new Date(tripInfo.trip.start) &&
+        today <= new Date(tripInfo.trip.end)
+      );
+    });
+    const planningTrips = myTripInfos.filter(tripInfo => {
```
Comment (rafiazman) [inline src/spa/components/homepage/UserHomePage.js:28]: this is backend functionality, assume array is always "correct"

File: src/spa/components/homepage/UserHomePage.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      return today < new Date(tripInfo.trip.start);
+    });
+    const pastTrips = myTripInfos.filter(tripInfo => {
```
Comment (rafiazman) [inline src/spa/components/homepage/UserHomePage.js:31]: this is backend functionality, assume array is always "correct"

File: src/spa/components/homepage/UserHomePage.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                        newChatNum={planningTrip.newNoteNum}
+                        newNoteNum={planningTrip.newChatNum}
+                      />
+                    </a>
+                  </Link>
+                ))}
+              </div>
+            ) : (
+              <h3>
+                No plans yet. <a>Plan for a trip</a> today!
```
Comment (rafiazman) [inline src/spa/components/homepage/UserHomePage.js:80]: good candidate for a UI test using Jest perhaps (snapshot testing maybe?)

Comment (rafiazman) [reply src/spa/components/homepage/UserHomePage.js:80]: no change required here, but would be great to have a test if possible

**Comment by rafiazman** [comment]
The user should be redirected to this page after they log in

**Comment by niuniuanran** [comment]
> The user should be redirected to this page after they log in, otherwise this looks pretty complete

Has been redirected

**Pull request merged**